pkg/util/dump/dump.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dump
+
+import (
+	"github.com/davecgh/go-spew/spew"
+)
+
+var prettyPrintConfig = &spew.ConfigState{
+	Indent:                  "  ",
+	DisableMethods:          true,
+	DisablePointerAddresses: true,
+	DisableCapacities:       true,
+}
+
+var prettyPrintConfigForHash = &spew.ConfigState{
```

Comment:
Some objs will be written to hash for comparation, so need some special config:https://github.com/kubernetes/kubernetes/blob/c8f001d798eb81a92e61e467fca74ca180de6fc6/pkg/util/hash/hash.go#L25-L37

Reply:
I am skeptical that the hash logic ever did was intended - without `DisablePointerAddresses` set, you can't hash two equivalent objects unless they have NO pointers in them.

So I want to say that we should change this to include `DisablePointerAddresses` and `DisableCapacities`, but I don't know what that would break.  There are not too many call-sites, so I think we should go through them all and verify either that I am wrong and it works, or I am right and changing it would not destroy things.

In particular the kubelet checkpoint is suspect to me.

Reply:
Sounds like `DisablePointerAddresses` and `DisableCapacities` don't work for `%#v`:
https://go.dev/play/p/pUjkEnixf9I

Reply:
Ahh!  I did not realize that.  Any reason not to set them to true in both cases, then?

Reply:
Will set